sycl/include/CL/sycl/handler.hpp

```diff
@@ -308,8 +307,8 @@ class __SYCL_EXPORT handler {
   /// Streams are then forwarded to command group and flushed in the scheduler.
   ///
   /// \param Stream is a pointer to SYCL stream.
-  void addStream(shared_ptr_class<detail::stream_impl> Stream) {
-    MStreamStorage.push_back(std::move(Stream));
+  void addStream(const shared_ptr_class<detail::stream_impl> &Stream) {
+    MStreamStorage.push_back(Stream);
```

Comment:
Not about your change: When I implemented handler::addReduction() adding objects to MSharedPtrStorage I thought that MStreamStorage could be deleted and MSHaredPtrStorage could be used instead. That would save the field without any bad consequences.

Reply:
Hm, it seems you are right. Will do.
I think refactoring patches are good candidates for "not related" comments.

Reply:
Understood, thank you.